### PR TITLE
fix(ReduxQuerySync): Allow unsubscribing from ReduxQuerySync

### PR DESCRIPTION
**Pull request by grafakus**
This prevents unpredictable behaviours when Pyroscope is embedded in a larger React application (e.g some URL query parameters could be reseted to their default value).

### CODE REVIEW

File: public/app/redux/redux-query-sync.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import { actions as tracingActions } from '@pyroscope/redux/reducers/tracing';
+import { history } from '@pyroscope/util/history';
+import ReduxQuerySync from 'redux-query-sync';
+import { actions as continuousActions } from './reducers/continuous';
+import { RootState, StoreType } from './store';
+
+let unsubscribeFromReduxQuerySync: unknown;
```
Comment (darrenjaneczek) [inline public/app/redux/redux-query-sync.ts:7]: I am not expecting that this should appear outside of react. Maybe we can make this into a hook, and the useEffect in the hook here can keep the `unsubscribeFromReduxQuerySync` and return it.

Then the `App` just needs to use the hook.

Comment (grafakus) [reply public/app/redux/redux-query-sync.ts:7]: Good point, I'll make the changes!

File: public/app/app.tsx [hunk 2]
```diff
@@ -1,26 +1,29 @@
 import React from 'react';
-import ReactDOM from 'react-dom/client';
+import { Sidebar } from '@pyroscope/components/Sidebar';
```
Comment (grafakus) [inline public/app/app.tsx:2]: Prettier...

File: public/app/redux/useReduxQuerySync.ts [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    },
+    initialTruth: 'location',
+    replaceState: false,
+    history,
+  });
+}
+
+export function useReduxQuerySync() {
+  useEffect(setupReduxQuerySync, []);
+}
```
Comment (darrenjaneczek) [inline public/app/redux/useReduxQuerySync.ts:104]: Tidy!

File: public/app/redux/useReduxQuerySync.ts [hunk 4]
```diff
@@ -0,0 +1,104 @@
+import { useEffect } from 'react';
+import { actions as tracingActions } from '@pyroscope/redux/reducers/tracing';
+import { history } from '@pyroscope/util/history';
+import ReduxQuerySync from 'redux-query-sync';
+import { actions as continuousActions } from './reducers/continuous';
+import store, { RootState } from './store';
+
+function setupReduxQuerySync() {
```
Comment (darrenjaneczek) [inline public/app/redux/useReduxQuerySync.ts:8]: I like how this isn't being exported.

**Review by darrenjaneczek — APPROVED** [review]
Excellent. Thanks!

**Pull request merged**